pkg/network/state.go

```diff
@@ -570,6 +609,10 @@ func (ns *networkState) storeClosedConnection(c *ConnectionStats) {
 			continue
 		}
 		client.closed.insert(c, ns.maxClosedConns)
+
+		if !client.closedConnectionsNearCapacity {
+			ns.updateClientCapacity(clientID, client)
```

Comment:
Do we need to update it here? We could just calculate this on demand.

Reply:
agreed, changed it to fully on demand and no extra state variables